modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ExecutionTest.java

```diff
@@ -1082,4 +1113,58 @@ else if (o2 != null)
             }
         };
     }
+
+    /**
+     * Node that always throws assertion error except for {@link #close()}
+     * and {@link #onRegister(Downstream)} methods.
+     */
+    static class CorruptedNode<T> implements Node<T> {
+        /** {@inheritDoc} */
+        @Override public ExecutionContext<T> context() {
+            assert false;
```

Comment:
throw new AssertionError, wold be preferable, -ea can be disabled on tests.

Reply:
sounds reasonable, fixed